assets/src/stories-editor/components/higher-order/with-call-to-action-validation.js

```diff
@@ -1,61 +1,64 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
 import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { isCTABlock } from '../../helpers';
 
-const enhance = compose(
-	/**
-	 * For blocks which are only allowed once per page,  provides the
-	 * wrapped component with `originalBlockClientId` -- a reference to the
-	 * first block of the same type on the page -- if and only if that
-	 * "original" block is not the current one. Thus, an non-existent
-	 * `originalBlockClientId` prop signals that the block is valid.
-	 */
-	withSelect( ( select, props ) => {
-		const { getBlockRootClientId, getBlock, getBlockOrder, getBlocksByClientId } = select( 'core/block-editor' );
-
-		if ( ! isCTABlock( props.name ) ) {
-			return {};
-		}
+export default createHigherOrderComponent( ( BlockEdit ) => {
+	const CallToActionValidation = ( props ) => {
+		const {
+			isInvalid,
+			originalBlockClientId,
+		} = useSelect( ( select ) => {
+			const { getBlockRootClientId, getBlock, getBlockOrder, getBlocksByClientId } = select( 'core/block-editor' );
 
-		const parentBlock = getBlock( getBlockRootClientId( props.clientId ) );
+			if ( ! isCTABlock( props.name ) ) {
+				return {};
+			}
 
-		if ( ! parentBlock ) {
-			return {};
-		}
+			const parentBlock = getBlock( getBlockRootClientId( props.clientId ) );
 
-		const isFirstPage = getBlockOrder().indexOf( parentBlock.clientId ) === 0;
-		const blocksOnPage = getBlocksByClientId( getBlockOrder( parentBlock.clientId ) );
-		const firstOfSameType = blocksOnPage.find( ( { name } ) => name === props.name );
-		const isInvalid = firstOfSameType && firstOfSameType.clientId !== props.clientId;
-
-		return {
-			isInvalid: isFirstPage || isInvalid,
-			originalBlockClientId: isInvalid && firstOfSameType.clientId,
-		};
-	} ),
-	withDispatch( ( dispatch, { originalBlockClientId } ) => ( {
-		selectFirst: () => dispatch( 'core/block-editor' ).selectBlock( originalBlockClientId ),
-	} ) ),
-);
+			if ( ! parentBlock ) {
+				return {};
+			}
+
+			const isFirstPage = getBlockOrder().indexOf( parentBlock.clientId ) === 0;
+			const blocksOnPage = getBlocksByClientId( getBlockOrder( parentBlock.clientId ) );
+			const firstOfSameType = blocksOnPage.find( ( { name } ) => name === props.name );
+			const _isInvalid = firstOfSameType && firstOfSameType.clientId !== props.clientId;
+
+			return {
+				isInvalid: isFirstPage || _isInvalid,
+				/**
+				 * This is a reference to the first block of the same type on the page -- if and only if that
+				 * "original" block is not the current one. Thus, an non-existent
+				 * `originalBlockClientId` prop signals that the block is valid.
+				 */
+				originalBlockClientId: _isInvalid && firstOfSameType.clientId,
+			};
+		}, [ props.name, props.clientId ] );
+
+		const { selectBlock } = useDispatch( 'core/block-editor' );
+		const selectFirst = useCallback(
```

Comment:
Does `selectBlock` need to be passed in here.

Reply:
As a dependency you mean? Yes. See https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
Useful to know :thinking: